### DISCUSSION TITLE
What is a first class function?

### DISCUSSION THREAD

**Discussion by montasim** [STRINGLABITSOLUTIONS/Bhojon#349] [Q&A] [answered]
In short

**Comment by d4tdev** [comment, marked as answer]
A programming language is said to have First-class functions when functions in that language are treated like any other variable. For example, in such a language, a function can be passed as an argument to other functions, can be returned by another function and can be assigned as a value to a variable.

First-class functions are functions that are treated as variables. They can also be parsed into other functions as arguments.

In JavaScript, functions are treated as values or just another type of object

**Comment by Preetiraj3697** [comment]
A programming language is said to have First-class functions when functions in that language are treated like any other variable. For example, in such a language, a function can be passed as an argument to other functions, can be returned by another function and can be assigned as a value to a variable.
const foo = () => {
  console.log("foobar");
};
foo(); // Invoke it using the variable
// foobar


[Passing a function as an argument](https://developer.mozilla.org/en-US/docs/Glossary/First-class_Function#passing_a_function_as_an_argument)
function sayHello() {
  return "Hello, ";
}
function greeting(helloMessage, name) {
  console.log(helloMessage() + name);
}
// Pass `sayHello` as an argument to `greeting` function
greeting(sayHello, "JavaScript!");
// Hello, JavaScript!
